### DISCUSSION TITLE
How to get chapter metadata to appear in PDF and EPUB of Quarto Book?

### DISCUSSION THREAD

**Discussion by IULibScholComm** [quarto-dev/quarto-cli#9382] [Q&A (Q1)] [answered]
### Description

How can chapter-specific metadata be made visible in the PDF and EPUB versions, as it is in the HTML version, of a Quarto book?

I ask as I'm hoping to create a Quarto book in which each chapter's metadata (author, doi, abstract, etc.) will be different from the book's metadata. While most of the different chapter and book metadata are made visible in the HTML version via the example below, very little of the different chapter and book metadata are made visible in the PDF and EPUB versions (only the different abstracts) via that example. I want, however, all the chapter and book metadata to be present in these versions to display such information as the different authors of each chapter, each chapter's doi, etc. Can this be done?

**Book Metadata**
````qmd
project:
  type: book

book:
  title: "Conference Proceedings"
  author: ["Norah Jones", "John Doe"]
  date: "4/12/2024"
  abstract: "This is a book of conference proceedings."
  doi: 10.5967/book

  chapters:
    - index.qmd
    - intro.qmd
    - summary.qmd
    - references.qmd

lang: en

language: 
  title-block-author-plural: "Editors"

bibliography: references.bib

format:
  html:
    theme: cosmo
  pdf:
    documentclass: scrreprt
````

**Chapter Metadata**
````qmd
---
author:
  - name: John Doe
    email: john@doe.edu
  - name: Jane Doe
    email: jane@doe.edu
abstract: >
  Essay about ...
doi: 10.5967/chapter
language:
  title-block-author-single: "Author"
  title-block-author-plural: "Authors"
---

# Conference Paper I

This is a book created from markdown and executable code.

See @knuth84 for additional discussion of literate programming.
````

**HTML Book Metadata**
![Book Metadata: HTML](https://github.com/quarto-dev/quarto-cli/assets/131798128/f3e626de-a22a-45b8-bc91-80e13f61fe64)

**HTML Chapter Metadata**
![Chapter Metadata: HTML](https://github.com/quarto-dev/quarto-cli/assets/131798128/bc3bce42-6b80-4ccf-88a1-dfc0018c7159)

**PDF Chapter Metadata Omits Chapter Author Name, DOI, etc.**
![Chapter Metadata: PDF--omits author name, doi, etc.](https://github.com/quarto-dev/quarto-cli/assets/131798128/16803333-05e6-4a0b-b4c3-1ca89e63d5d4)

**EPUB Chapter Metadata Omits Chapter Author Name, DOI, etc.**
![Chapter Metadata: EPUB--omits author name, doi, etc.](https://github.com/quarto-dev/quarto-cli/assets/131798128/56ea2de1-a2d7-464e-bb42-d397d79f31bf)

**Comment by mcanouil** [comment, marked as answer]
You're gonna need to use a Lua filter to inject the content where you want it.
In the case of HTML, it's simpler because it's done via the title block partial which is used on all pages, but for PDF, there are no pages/documents. It's one single document.

**Comment by janpfander** [comment]
By any chance, has someone come up with a lua filter to achieve displaying chapter-specific meta-data in pdf quarto books?

**Comment by biplovbhandari** [comment]
My workaround: a content-visible when-format="pdf" markdown cell in each chapter notebook, right after the chapter heading, with author names and ORCID icons:
```
::: {.content-visible when-format="pdf"}
**Authors:** Author One `\orcidlink{0000-0001-...}`{=latex}, Author Two
:::
```
Not elegant, but works reliably. Has anyone since implemented a Lua filter that reads per-chapter author: metadata and injects it automatically in PDF output?

https://github.com/NASA-EarthRISE/EarthRISE-Applied-Artificial-Intelligence-and-Deep-Learning-Book/issues/38
